firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionGenerator.kt

```diff
@@ -36,35 +36,25 @@ internal data class SessionDetails(
  *
  * @hide
  */
-internal class SessionGenerator(private var collectEvents: Boolean) {
-  private var firstSessionId = ""
-  private var sessionIndex: Int = -1
+internal class SessionGenerator(private val collectEvents: Boolean) {
+  private val firstSessionId = generateSessionId()
+  private var sessionIndex = 0
 
-  private var thisSession: SessionDetails =
-    SessionDetails(
-      sessionId = "",
-      firstSessionId = "",
-      collectEvents,
-      sessionIndex,
-    )
+  var currentSession =
+    SessionDetails(sessionId = "", firstSessionId = "", collectEvents, sessionIndex = -1)
```

Comment:
Do we still need this default instance? Would it make sense to make currentSession nullable or lateinit?

Reply:
It mostly depends on preference. Do you want to deal with nulls in the classes that use this, or assume that what you get is valid? 

I think we've designed this class to always return a valid currentSession, so I lean towards keeping it not nullable.

Reply:
I don't think we need a default instance. That does not serve anything. SessionGenerator should generate session only when there is a need.